### PR TITLE
TC-1096-roles-and-permissions-saving-a-permafrost-role-in

### PR DESCRIPTION
**Pull request by devo-wm**

### CODE REVIEW

File: permafrost/views.py [hunk 1]
```diff
@@ -261,6 +261,11 @@ def get_context_data(self, **kwargs):
         )
         return context
 
+    def get_form_kwargs(self):
```
Comment (devo-wm) [inline permafrost/views.py:264]: Here's the actual fix!

File: permafrost/tests.py [hunk 2]
```diff
@@ -505,6 +506,19 @@ def test_role_update_POST_updates_name(self):
         updated_role = PermafrostRole.objects.get(pk=self.pf_role.pk)
         self.assertEqual(updated_role.name, "Test Change")
     
+    def test_role_update_POST_updates_when_no_values_are_changed(self):
+        uri = reverse('permafrost:role-update', kwargs={'slug': 'test-role'})
```
Comment (devo-wm) [inline permafrost/tests.py:510]: This test triggers the integrity errror before I added the get_form_kwargs method below

File: permafrost/tests.py [hunk 3]
```diff
@@ -298,9 +298,10 @@ class PermafrostViewTests(TestCase):
 
     def setUp(self):
         self.client = Client()
-        self.pf_role = PermafrostRole.objects.create(category="staff", name="Test Role")
-        super_user = get_user_model().objects.get(pk=1)
-        self.client.force_login(super_user)
+        self.pf_role = PermafrostRole.objects.create(category="staff", name="Test Role", site=Site.objects.get_current())
+        PermafrostRole.objects.create(category="staff", name="Test Role", site=Site.objects.get(pk=2))
```
Comment (devo-wm) [inline permafrost/tests.py:302]: Added a matching name on a different site to trigger the integrity error.

File: permafrost/views.py [hunk 4]
```diff
@@ -184,6 +184,12 @@ def get_form_class(self):
             return SelectPermafrostRoleTypeForm
         return PermafrostRoleCreateForm
 
+    def get_form_kwargs(self):
```
Comment (devo-wm) [inline permafrost/views.py:187]: And here's the fix for https://whitemoondreams.atlassian.net/browse/TC-1052

**Review by mgcamisa — APPROVED** [review]

**Pull request merged**